.gitignore

```diff
@@ -27,6 +27,24 @@ CMakeFiles/
 # don't ignore cism build utilities directory
 !components/cism/**/build
 
+# configure / build files
```

Comment:
I assume that all of these new entries are referring to files that appear in specific places in the directory tree? If so, the rules should be more specific, listing the exact paths where these files occur.

Also: I started doing this by having multiple `.gitignore` files in different places. However, I was just doing some googling, and it sounds like the most common convention is to put everything in this top-level .gitignore file. So I'd like to rework things to do that. So, for example, this entry in `src/unit_test_stubs/utils/.gitignore`: `restUtilMod_stub.F90` would go in the top-level as `/src/unit_test_stubs/utils/restUtilMod_stub.F90` (I think it's correct to have this leading slash here: I think this says to match exactly this path). I can do this myself in a following PR if you don't want to deal with it.